rclcpp/include/rclcpp/executors/executor_entities_collection.hpp

```diff
@@ -72,34 +73,47 @@ void update_entities(
   std::function<void(const typename CollectionType::EntitySharedPtr &)> on_removed
 )
 {
-  for (auto it = update_to.begin(); it != update_to.end(); ) {
+  for (auto it = update_to.begin_ordered(); it != update_to.end_ordered(); ) {
     if (update_from.count(it->first) == 0) {
       auto entity = it->second.entity.lock();
       if (entity) {
         on_removed(entity);
       }
-      it = update_to.erase(it);
+      it = update_to.erase_ordered(it);
     } else {
       ++it;
     }
   }
-  for (auto it = update_from.begin(); it != update_from.end(); ++it) {
+  for (auto it = update_from.begin_ordered(); it != update_from.end_ordered(); ++it) {
     if (update_to.count(it->first) == 0) {
       auto entity = it->second.entity.lock();
       if (entity) {
         on_added(entity);
       }
-      update_to.insert(*it);
+      bool inserted = update_to.insert(*it);
+      // Should never be false, so this is a defensive check, mark unused too
+      // in order to avoid a warning in release builds.
+      assert(inserted);
+      RCUTILS_UNUSED(inserted);
```

Comment:
This is the important part of the changes that affect the order of execution when tied, because this is what controls the order in which they are added to the wait set in the case of the wait set based executors, i.e. everything except the `EventsExecutor`.